expr/parse.go

```diff
@@ -337,21 +337,33 @@ func parseString(s string) (string, string, error) {
 	return s[:i], s[i+1:], nil
 }
 
-// exctractMetric searches for a metric name in `m'
+// extractMetric searches for a metric name in `m'
 // metric name is defined to be a series of name characters terminated by a comma
```

Comment:
this function has never been well documented, but TestExtractMetric makes up for a lot. the changes you're doing warrant updates to TestExtractMetric though, just so that's clearer what the extra stuff is for
(and maybe it's time to just document this function)